Specification for EXMARaLDA import?

**Is your feature request related to a problem? Please describe.**
I would like to understand how to import EXMARaLDA files into Hexatomic. Right now, my naive attempts (with a transcription from the EXMARaLDA demo-corpus) fail with error messages of the following kind in the log:

```
08:18:25.732 [EXMARaLDAImporter_mapper(salt:/EXMARaLDA_DemoKorpus_EN/StructureErrors)] ERROR MyImporter - An exception was thrown by the mapper threads 'Thread[EXMARaLDAImporter_mapper(salt:/EXMARaLDA_DemoKorpus_EN/StructureErrors),5,EXMARaLDAImporter_mapperGroup]'.
org.corpus_tools.pepper.modules.exceptions.PepperModuleDataException: A data error occured for a Pepper mapper 'org.corpus_tools.peppermodules.exmaralda.EXMARaLDA2SaltMapper'. Cannot convert given exmaralda file 'file:/C:/Users/bernd/OneDrive/Desktop/EXMARaLDA_DemoKorpus_EN/StructureErrors.xml', because no textual source layer was found.
        at org.corpus_tools.peppermodules.exmaralda.EXMARaLDA2SaltMapper.mapSDocument(EXMARaLDA2SaltMapper.java:217)
        at org.corpus_tools.pepper.impl.PepperMapperControllerImpl.map(PepperMapperControllerImpl.java:251)
        at org.corpus_tools.pepper.impl.PepperMapperControllerImpl.run(PepperMapperControllerImpl.java:188)
08:18:26.540 [ModalContext] ERROR o.c.h.f.i.ImportWizard - Error(s) during import: Corpus was not imported due to errors. Please check the log messages for any issues and make sure you have selected the correct format.
```

```
08:21:32.141 [EXMARaLDAImporter_mapper(salt:/PaulMcCartney/PaulMcCartney)] ERROR MyImporter - Cannot map 'salt:/0/PaulMcCartney/PaulMcCartney' with module 'EXMARaLDAImporter', because of a mapping result was 'FAILED'.
08:21:32.142 [EXMARaLDAImporter_mapper(salt:/PaulMcCartney/PaulMcCartney)] ERROR MyImporter - An exception was thrown by the mapper threads 'Thread[EXMARaLDAImporter_mapper(salt:/PaulMcCartney/PaulMcCartney),5,EXMARaLDAImporter_mapperGroup]'.
org.corpus_tools.pepper.modules.exceptions.PepperModuleException: Any exception occured while mapping.
        at org.corpus_tools.pepper.impl.PepperMapperControllerImpl.run(PepperMapperControllerImpl.java:193)
Caused by: org.corpus_tools.salt.exceptions.SaltException: Cannot set the name of this label object, because it is empty.
        at org.corpus_tools.salt.graph.impl.LabelImpl.setName(LabelImpl.java:96)
        at org.corpus_tools.hexatomic.core.events.salt.LabelNotifierImpl.lambda$4(LabelNotifierImpl.java:84)
        at org.corpus_tools.hexatomic.core.events.salt.LabelNotifierImpl.modified(LabelNotifierImpl.java:112)
        at org.corpus_tools.hexatomic.core.events.salt.LabelNotifierImpl.setName(LabelNotifierImpl.java:84)
        at org.corpus_tools.salt.graph.impl.LabelImpl.setName(LabelImpl.java:92)
        at org.corpus_tools.peppermodules.exmaralda.EXMARaLDA2SaltMapper.mapUDInformations2SAnnotationContainer(EXMARaLDA2SaltMapper.java:472)
        at org.corpus_tools.peppermodules.exmaralda.EXMARaLDA2SaltMapper.mapMetaInformation2SDocument(EXMARaLDA2SaltMapper.java:372)
        at org.corpus_tools.peppermodules.exmaralda.EXMARaLDA2SaltMapper.mapSDocument(EXMARaLDA2SaltMapper.java:172)
        at org.corpus_tools.pepper.impl.PepperMapperControllerImpl.map(PepperMapperControllerImpl.java:251)
        at org.corpus_tools.pepper.impl.PepperMapperControllerImpl.run(PepperMapperControllerImpl.java:188)
08:21:33.118 [ModalContext] ERROR o.c.h.f.i.ImportWizard - Error(s) during import: Corpus was not imported due to errors. Please check the log messages for any issues and make sure you have selected the correct format.
```
The first message seems to be caused because the importer is also attempting to read a "*.xml" file which is not an EXMARaLDA transcription. The second message, I cannot interpret. 

**Describe the solution you'd like**
I would like a specification of what properties an EXB file should have and how the folder structure of the corpus should be organised so that it can be imported into Hexatomic.

**Describe alternatives you've considered**
Not importing it ;-)

**Additional context**


**Get credited**
@berndmoos

--- a/.classpath
+++ b/.classpath
@@ -36,22 +36,5 @@
 			<attribute name="maven.pomderived" value="true"/>
 		</attributes>
 	</classpathentry>
-	<classpathentry kind="src" path="target/generated-sources/annotations">
-		<attributes>
-			<attribute name="optional" value="true"/>
-			<attribute name="maven.pomderived" value="true"/>
-			<attribute name="ignore_optional_problems" value="true"/>
-			<attribute name="m2e-apt" value="true"/>
-		</attributes>
-	</classpathentry>
-	<classpathentry kind="src" output="target/test-classes" path="target/generated-test-sources/test-annotations">
-		<attributes>
-			<attribute name="optional" value="true"/>
-			<attribute name="maven.pomderived" value="true"/>
-			<attribute name="ignore_optional_problems" value="true"/>
-			<attribute name="m2e-apt" value="true"/>
-			<attribute name="test" value="true"/>
-		</attributes>
-	</classpathentry>
 	<classpathentry kind="output" path="target/classes"/>
 </classpath>

--- a/README.md
+++ b/README.md
@@ -87,7 +87,7 @@ corresponding to the xml representation:
 Those data are mapped to a single span overlapping the 4 tokens in Salt. The span further is annotated with 'SPK1::s=sentence'.
 The mapping of the several groups of transcriptions and annotations defined by different speakers are processed independently. So the mapping of multiple speakers with several annotations is analog to one speaker.
 
-##Audio files
+## Audio files
 
 The mapping uses the audio file passed with the model in the xml-element referenced-file:
 
@@ -111,6 +111,7 @@ If an audio file is given the mapping creates a audio data source in Salt (SAudi
 In this case, the first token 'Hello' is aligned to the range [0.0, 0.123] in the audio file in Salt. 
 
 ## Properties
+
 The table contains an overview of all usable properties to customize the behavior of this Pepper module. 
 
 |Name of property    |Type of property	                                            |optional/ mandatory |  default value |
@@ -127,6 +128,7 @@ The table contains an overview of all usable properties to customize the behavio
 |mapTimeline         |Boolean	                                                    |optional            |    true |
 |mapDescriptions     |Boolean	                                                    |optional            |    true |
 |mapSpeakerMetadata  |Boolean	                                                    |optional            |    true |
+|strictFileType  |Boolean	                                                        |optional            |    true |
 
 ### salt.tokenSeparator
 With the property salt.tokenSeparatoryou can set a single sign or a sequence of signs which shall be used between the concatenation of event values representing the primary data. 
@@ -212,16 +214,26 @@ Setting this property to true, adds an additional event in tier 'primary data' c
 ### trimToken
 When this property is true, all tokens defined in exmaralda are trimmed before they are mapped to Salt. That means for instance trailing blanks are removed.
 
-## mapTimeline
+### mapTimeline
 When this property is set to `false`, the original timeline in the input file will not be imported. The default is to include the timeline. 
 
-## mapDescriptions
+### mapDescriptions
 
 Map tiers of the type "Description" if set to  "true", which is the default value.
 
-## mapSpeakerMetadata
+### mapSpeakerMetadata
 
-Import the metadata of the speaker if set to "true", which is the default value.s
+Import the metadata of the speaker if set to "true", which is the default value.
+
+### strictFileType
+
+The default setting of `true` restricts importing only files with the ending `.exb`.
+Older versions of the importer searched for files with the ending `.exb`,
+`.xml`, `.xmi` and `.exmaralda`.
+But the EXMARaLDA partitur editor only creates files with the ending `.exb` and
+only these files should be imported.
+Set to `false` for getting the old behavior.
+
 
 # <a name="exporter">EXMARaLDAExporter</a>
 

--- a/pom.xml
+++ b/pom.xml
@@ -6,7 +6,7 @@
 		<version>3.3.3</version>
 	</parent>
 	<artifactId>pepperModules-EXMARaLDAModules</artifactId>
-	<version>1.3.1-SNAPSHOT</version>
+	<version>2.0.0-SNAPSHOT</version>
 	<packaging>bundle</packaging>
 	<issueManagement>
 		<system>GitHub</system>

--- a/src/main/java/org/corpus_tools/peppermodules/exmaralda/EXMARaLDAImporter.java
+++ b/src/main/java/org/corpus_tools/peppermodules/exmaralda/EXMARaLDAImporter.java
@@ -1,27 +1,27 @@
 /**
  * Copyright 2009 Humboldt-Universität zu Berlin, INRIA.
  *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
  *
- *       http://www.apache.org/licenses/LICENSE-2.0
+ * http://www.apache.org/licenses/LICENSE-2.0
  *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
  *
  *
  */
 package org.corpus_tools.peppermodules.exmaralda;
 
+import de.hu_berlin.german.korpling.saltnpepper.misc.exmaralda.resources.EXBResourceFactory;
 import org.corpus_tools.pepper.common.PepperConfiguration;
 import org.corpus_tools.pepper.core.SelfTestDesc;
 import org.corpus_tools.pepper.impl.PepperImporterImpl;
 import org.corpus_tools.pepper.modules.PepperImporter;
 import org.corpus_tools.pepper.modules.PepperMapper;
+import org.corpus_tools.salt.SALT_TYPE;
 import org.corpus_tools.salt.common.SDocument;
 import org.corpus_tools.salt.graph.Identifier;
 import org.eclipse.emf.common.util.URI;
@@ -30,85 +30,111 @@ import org.eclipse.emf.ecore.resource.impl.ResourceSetImpl;
 import org.eclipse.emf.ecore.xmi.impl.XMIResourceFactoryImpl;
 import org.osgi.service.component.annotations.Component;
 
-import de.hu_berlin.german.korpling.saltnpepper.misc.exmaralda.resources.EXBResourceFactory;
-
 @Component(name = "EXMARaLDAImporterJavaComponent", factory = "PepperImporterComponentFactory")
 public class EXMARaLDAImporter extends PepperImporterImpl implements PepperImporter {
-	public static final String[] EXMARALDA_FILE_ENDINGS = { "exb", "xml", "xmi", "exmaralda" };
+  public static final String[] EXMARALDA_FILE_ENDINGS = {"exb", "xml", "xmi", "exmaralda"};
 
-	public EXMARaLDAImporter() {
-		super();
-		// setting name of module
-		setName("EXMARaLDAImporter");
-		setSupplierContact(URI.createURI(PepperConfiguration.EMAIL));
-		setSupplierHomepage(URI.createURI("https://github.com/korpling/pepperModules-EXMARaLDAModules"));
-		setDesc("This importer transforms data in the exb format of EXMARaLDA to a Salt model. ");
-		setProperties(new EXMARaLDAImporterProperties());
-		// set list of formats supported by this module
-		addSupportedFormat("EXMARaLDA", "1.0", null);
+  public EXMARaLDAImporter() {
+    super();
+    // setting name of module
+    setName("EXMARaLDAImporter");
+    setSupplierContact(URI.createURI(PepperConfiguration.EMAIL));
+    setSupplierHomepage(
+        URI.createURI("https://github.com/korpling/pepperModules-EXMARaLDAModules"));
+    setDesc("This importer transforms data in the exb format of EXMARaLDA to a Salt model. ");
+    setProperties(new EXMARaLDAImporterProperties());
+    // set list of formats supported by this module
+    addSupportedFormat("EXMARaLDA", "1.0", null);
 
-		// adding all file endings to list of endings for documents (necessary
-		// for importCorpusStructure)
-		for (String ending : EXMARALDA_FILE_ENDINGS) {
-			getDocumentEndings().add(ending);
-		}
-	}
+    // adding all file endings to list of endings for documents (necessary
+    // for importCorpusStructure)
+    for (String ending : EXMARALDA_FILE_ENDINGS) {
+      getDocumentEndings().add(ending);
+    }
+  }
 
-	/** emf resource loader **/
-	private ResourceSet resourceSet = null;
+  /** emf resource loader **/
+  private ResourceSet resourceSet = null;
 
-	private ResourceSet getResourceSet() {
-		if (resourceSet == null) {
-			synchronized (this) {
-				if (resourceSet == null) {
-					resourceSet = new ResourceSetImpl();
-					// Register XML resource factory
-					resourceSet.getResourceFactoryRegistry().getExtensionToFactoryMap().put("exmaralda",
-							new XMIResourceFactoryImpl());
-					resourceSet.getResourceFactoryRegistry().getExtensionToFactoryMap().put("xmi",
-							new XMIResourceFactoryImpl());
-					resourceSet.getResourceFactoryRegistry().getExtensionToFactoryMap().put("exb",
-							new EXBResourceFactory());
-					resourceSet.getResourceFactoryRegistry().getExtensionToFactoryMap().put("xml",
-							new EXBResourceFactory());
-				}
+  private ResourceSet getResourceSet() {
+    if (resourceSet == null) {
+      synchronized (this) {
+        if (resourceSet == null) {
+          resourceSet = new ResourceSetImpl();
+          // Register XML resource factory
+          resourceSet.getResourceFactoryRegistry().getExtensionToFactoryMap().put("exmaralda",
+              new XMIResourceFactoryImpl());
+          resourceSet.getResourceFactoryRegistry().getExtensionToFactoryMap().put("xmi",
+              new XMIResourceFactoryImpl());
+          resourceSet.getResourceFactoryRegistry().getExtensionToFactoryMap().put("exb",
+              new EXBResourceFactory());
+          resourceSet.getResourceFactoryRegistry().getExtensionToFactoryMap().put("xml",
+              new EXBResourceFactory());
+        }
 
-			}
-		}
-		return (resourceSet);
-	}
+      }
+    }
+    return (resourceSet);
+  }
 
-	@Override
-	public SelfTestDesc getSelfTestDesc() {
-		return new SelfTestDesc(
-				getResources().appendSegment("selfTests").appendSegment("exmaraldaImporter").appendSegment("in"),
-				getResources().appendSegment("selfTests").appendSegment("exmaraldaImporter").appendSegment("expected"));
-	}
+  @Override
+  public SelfTestDesc getSelfTestDesc() {
+    return new SelfTestDesc(
+        getResources().appendSegment("selfTests").appendSegment("exmaraldaImporter")
+            .appendSegment("in"),
+        getResources().appendSegment("selfTests").appendSegment("exmaraldaImporter")
+            .appendSegment("expected"));
+  }
 
-	@Override
-	public Double isImportable(URI corpusPath) {
-		Double retValue = 0.0;
-		for (String content : sampleFileContent(corpusPath, EXMARALDA_FILE_ENDINGS)) {
-			if ((content.contains("<?xml version=\"1.0\"")) && (content.contains("<basic-transcription>"))) {
-				retValue = 1.0;
-				break;
-			}
-		}
-		return retValue;
-	}
-	
-	/**
-	 * Creates a mapper of type {@link EXMARaLDA2SaltMapper}.
-	 * {@inheritDoc PepperModule#createPepperMapper(Identifier)}
-	 */
-	@Override
-	public PepperMapper createPepperMapper(Identifier sElementId) {
-		EXMARaLDA2SaltMapper mapper = new EXMARaLDA2SaltMapper();
-		URI resourcePath = getIdentifier2ResourceTable().get(sElementId);
-		if (sElementId.getIdentifiableElement() instanceof SDocument) {
-			mapper.setResourceURI(resourcePath);
-			mapper.setResourceSet(getResourceSet());
-		}
-		return (mapper);
-	}
+  @Override
+  public Double isImportable(URI corpusPath) {
+    Double retValue = 0.0;
+    for (String content : sampleFileContent(corpusPath, EXMARALDA_FILE_ENDINGS)) {
+      if ((content.contains("<?xml version=\"1.0\""))
+          && (content.contains("<basic-transcription>"))) {
+        retValue = 1.0;
+        break;
+      }
+    }
+    return retValue;
+  }
+
+
+  @Override
+  public SALT_TYPE setTypeOfResource(URI resource) {
+    SALT_TYPE result = super.setTypeOfResource(resource);
+    if (result == SALT_TYPE.SDOCUMENT && getProps().getStrictFileType()) {
+      // Only allow "exb" file ending
+      if ("exb".equalsIgnoreCase(resource.fileExtension())) {
+        return SALT_TYPE.SDOCUMENT;
+
+      } else {
+        return null;
+      }
+    } else {
+      return result;
+    }
+  }
+
+  /**
+   * Creates a mapper of type {@link EXMARaLDA2SaltMapper}.
+   * {@inheritDoc PepperModule#createPepperMapper(Identifier)}
+   */
+  @Override
+  public PepperMapper createPepperMapper(Identifier sElementId) {
+    EXMARaLDA2SaltMapper mapper = new EXMARaLDA2SaltMapper();
+    URI resourcePath = getIdentifier2ResourceTable().get(sElementId);
+    if (sElementId.getIdentifiableElement() instanceof SDocument) {
+      mapper.setResourceURI(resourcePath);
+      mapper.setResourceSet(getResourceSet());
+    }
+    return (mapper);
+  }
+
+  /**
+   * casts {@link PepperModulePropertiesImpl} to {@link EXMARaLDAImporterProperties}
+   **/
+  public EXMARaLDAImporterProperties getProps() {
+    return ((EXMARaLDAImporterProperties) this.getProperties());
+  }
 }

--- a/src/main/java/org/corpus_tools/peppermodules/exmaralda/EXMARaLDAImporterProperties.java
+++ b/src/main/java/org/corpus_tools/peppermodules/exmaralda/EXMARaLDAImporterProperties.java
@@ -1,17 +1,15 @@
 /**
  * Copyright 2009 Humboldt-Universität zu Berlin, INRIA.
  *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
  *
- *       http://www.apache.org/licenses/LICENSE-2.0
+ * http://www.apache.org/licenses/LICENSE-2.0
  *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
  *
  *
  */
@@ -23,7 +21,6 @@ import java.util.Map;
 import java.util.Set;
 import java.util.regex.Matcher;
 import java.util.regex.Pattern;
-
 import org.corpus_tools.pepper.modules.PepperModuleProperties;
 import org.corpus_tools.pepper.modules.PepperModuleProperty;
 import org.corpus_tools.pepper.modules.exceptions.PepperModuleException;
@@ -41,195 +38,248 @@ import org.slf4j.LoggerFactory;
  */
 @SuppressWarnings("serial")
 public class EXMARaLDAImporterProperties extends PepperModuleProperties {
-	private static final Logger logger = LoggerFactory.getLogger(EXMARaLDAImporter.class);
-	
-	@Deprecated
-	public static final String PROP_TOKEN_TIER = "salt.token";
-	public static final String PROP_TOKENSEP = "salt.tokenSeparator";
-	public static final String PROP_TIERMERGE = "salt.tierMerge";
-	public static final String PROP_LAYERS_SMALL = "salt.layers";
-	public static final String PROP_LAYERS_BIG = "salt.Layers";
-	public static final String PROP_URI_ANNOTATION = "salt.URIAnnotation";
-	public static final String PROP_SALT_SEMANTICS_POS = "saltSemantics.POS";
-	public static final String PROP_SALT_SEMANTICS_LEMMA = "saltSemantics.LEMMA";
-	public static final String PROP_SALT_SEMANTICS_WORD = "saltSemantics.WORD";
-	public static final String PROP_CLEAN_MODEL = "cleanModel";
-	/** When this property is true, all tokens defined in exmaralda are trimmed before they are mapped to Salt. That means for instance trailing blanks are removed. **/
-	public static final String PROP_TRIM_EVENTS = "trimEvents";
-	public static final String PROP_PARSE_NAMESPACE = "parseNamespace";
-	public static final String PROP_MAP_TIMELINE = "mapTimeline";
-	public static final String PROP_MAP_DESCRIPTIONS = "mapDescriptions";
-	public static final String PROP_MAP_SPEAKER_METADATA = "mapSpeakerMetadata";
-	public static final String PROP_TEXT_NAME_WITH_NAMESPACE = "qualifyTextNames";
-	
+  private static final Logger logger = LoggerFactory.getLogger(EXMARaLDAImporter.class);
 
-	public EXMARaLDAImporterProperties() {
-		this.addProperty(new PepperModuleProperty(PROP_TOKEN_TIER, String.class, "With this property you can mark the Tier object which shall be interpreted as the one containing the tokenization and the primary text.", false));
-		this.addProperty(new PepperModuleProperty<String>(PROP_TOKENSEP, String.class, "With this property you can set a single sign or a sequence of signs which shall be used between the concatenation of event values representing the primary data.", false));
-		this.addProperty(new PepperModuleProperty<String>(PROP_TIERMERGE, String.class, "In EXMARaLDA there is just one value per each Event object. When mapping those data to Salt, this restriction results, that a SSpan or SToken object will be created for each Event object. To avoid this, you can mark a sequence of sets of tiers to map them to a single SSpan or SToken object in Salt.", false));
-		this.addProperty(new PepperModuleProperty<String>(PROP_LAYERS_SMALL, String.class, "With this property you can use the layer mechanism of Salt. This means, with this property, you can map all SToken or SSpan object caused by Event objects to an SLayer object. You can also group SToken or SSpan coming from Event objects contained in several Tier objects to the same SLayer object.", false));
-		this.addProperty(new PepperModuleProperty<String>(PROP_LAYERS_BIG, String.class, "With this property you can use the layer mechanism of Salt. This means, with this property, you can map all SToken or SSpan object caused by Event objects to an SLayer object. You can also group SToken or SSpan coming from Event objects contained in several Tier objects to the same SLayer object.", false));
-		this.addProperty(new PepperModuleProperty<String>(PROP_URI_ANNOTATION, String.class, "With the property salt.URIAnnotation, you can mark a Tier object containing annotations, which are references in URI notation and are not simple String values. When mapping the data to Salt the resulting SAnnotation.sValue objects will be typed as URI values, so that each interpreting tool can interpret them as references.", false));
-		this.addProperty(new PepperModuleProperty<String>(PROP_SALT_SEMANTICS_POS, String.class, "You can influence the creation of objects in Salt to have a more semantic typing when mapping data to Salt. Here we provide three properties which can be used for a closer definition or typing of SAnnotation, SToken or SSpan objects conform to ISOCat1. This can be important in the case of a further processing with Pepper. Some modules exist, which only can deal with semantical enriched data for example they need a special kind of annotation like part-of-speech for their processing.", false));
-		this.addProperty(new PepperModuleProperty<String>(PROP_SALT_SEMANTICS_LEMMA, String.class, "You can influence the creation of objects in Salt to have a more semantic typing when mapping data to Salt. Here we provide three properties which can be used for a closer definition or typing of SAnnotation, SToken or SSpan objects conform to ISOCat1. This can be important in the case of a further processing with Pepper. Some modules exist, which only can deal with semantical enriched data for example they need a special kind of annotation like part-of-speech for their processing.", false));
-		this.addProperty(new PepperModuleProperty<String>(PROP_SALT_SEMANTICS_WORD, String.class, "You can influence the creation of objects in Salt to have a more semantic typing when mapping data to Salt. Here we provide three properties which can be used for a closer definition or typing of SAnnotation, SToken or SSpan objects conform to ISOCat1. This can be important in the case of a further processing with Pepper. Some modules exist, which only can deal with semantical enriched data for example they need a special kind of annotation like part-of-speech for their processing.", false));
-		this.addProperty(new PepperModuleProperty<Boolean>(PROP_CLEAN_MODEL, Boolean.class, "....", false, false));
-		this.addProperty(new PepperModuleProperty<Boolean>(PROP_TRIM_EVENTS, Boolean.class, "When this property is true, all events in exmaralda are trimmed before they are mapped to Salt. That means for instance trailing blanks are removed. ", false, false));
-		this.addProperty(new PepperModuleProperty<Boolean>(PROP_PARSE_NAMESPACE, Boolean.class, "Parse the qualified annotation name including the namespace (e.g. myns::annoname) from the category name.", false, false));
-		this.addProperty(new PepperModuleProperty<Boolean>(PROP_MAP_TIMELINE, Boolean.class, "Import the EXMARaLDA timeline as Salt STimeline", true, false));
-		this.addProperty(PepperModuleProperty.create().withName(PROP_MAP_DESCRIPTIONS).withType(Boolean.class).withDescription("Map tiers of the type \\\"Description\\\"").withDefaultValue(true).build());
-		this.addProperty(PepperModuleProperty.create().withName(PROP_MAP_SPEAKER_METADATA).withType(Boolean.class).withDescription("Import the metadata of the speaker").withDefaultValue(true).build());
-		this.addProperty(PepperModuleProperty.create().withName(PROP_TEXT_NAME_WITH_NAMESPACE).withType(Boolean.class).withDescription("If true, texts will get qualified names with speaker name as namespace. If false, no namespace is used.").withDefaultValue(false).build());
-	}
+  @Deprecated
+  public static final String PROP_TOKEN_TIER = "salt.token";
+  public static final String PROP_TOKENSEP = "salt.tokenSeparator";
+  public static final String PROP_TIERMERGE = "salt.tierMerge";
+  public static final String PROP_LAYERS_SMALL = "salt.layers";
+  public static final String PROP_LAYERS_BIG = "salt.Layers";
+  public static final String PROP_URI_ANNOTATION = "salt.URIAnnotation";
+  public static final String PROP_SALT_SEMANTICS_POS = "saltSemantics.POS";
+  public static final String PROP_SALT_SEMANTICS_LEMMA = "saltSemantics.LEMMA";
+  public static final String PROP_SALT_SEMANTICS_WORD = "saltSemantics.WORD";
+  public static final String PROP_CLEAN_MODEL = "cleanModel";
+  /**
+   * When this property is true, all tokens defined in exmaralda are trimmed before they are mapped
+   * to Salt. That means for instance trailing blanks are removed.
+   **/
+  public static final String PROP_TRIM_EVENTS = "trimEvents";
+  public static final String PROP_PARSE_NAMESPACE = "parseNamespace";
+  public static final String PROP_MAP_TIMELINE = "mapTimeline";
+  public static final String PROP_MAP_DESCRIPTIONS = "mapDescriptions";
+  public static final String PROP_MAP_SPEAKER_METADATA = "mapSpeakerMetadata";
+  public static final String PROP_TEXT_NAME_WITH_NAMESPACE = "qualifyTextNames";
+  public static final String PROP_STRICT_FILE_TYPE = "strictFileType";
 
-	@Deprecated
-	public Set<String> getTokenTiers() {
-		String prop = ((String) this.getProperty(PROP_TOKEN_TIER).getValue());
-		Set<String> tokenTiers = new LinkedHashSet<String>();
-		if ((prop != null) && (!prop.isEmpty())) {
-			if (prop.startsWith("{")) {
-				prop = prop.replace("{", "").replace("}", "");
-				String[] splitted = prop.split(",");
-				for (String s : splitted) {
-					tokenTiers.add(s.trim());
-				}
-			} else {
-				tokenTiers.add(prop.trim());
-			}
-		}
-		return (tokenTiers);
-	}
 
-	public String getTokenSeparator() {
-		return ((String) this.getProperty(PROP_TOKENSEP).getValue());
-	}
+  public EXMARaLDAImporterProperties() {
+    this.addProperty(new PepperModuleProperty(PROP_TOKEN_TIER, String.class,
+        "With this property you can mark the Tier object which shall be interpreted as the one containing the tokenization and the primary text.",
+        false));
+    this.addProperty(new PepperModuleProperty<String>(PROP_TOKENSEP, String.class,
+        "With this property you can set a single sign or a sequence of signs which shall be used between the concatenation of event values representing the primary data.",
+        false));
+    this.addProperty(new PepperModuleProperty<String>(PROP_TIERMERGE, String.class,
+        "In EXMARaLDA there is just one value per each Event object. When mapping those data to Salt, this restriction results, that a SSpan or SToken object will be created for each Event object. To avoid this, you can mark a sequence of sets of tiers to map them to a single SSpan or SToken object in Salt.",
+        false));
+    this.addProperty(new PepperModuleProperty<String>(PROP_LAYERS_SMALL, String.class,
+        "With this property you can use the layer mechanism of Salt. This means, with this property, you can map all SToken or SSpan object caused by Event objects to an SLayer object. You can also group SToken or SSpan coming from Event objects contained in several Tier objects to the same SLayer object.",
+        false));
+    this.addProperty(new PepperModuleProperty<String>(PROP_LAYERS_BIG, String.class,
+        "With this property you can use the layer mechanism of Salt. This means, with this property, you can map all SToken or SSpan object caused by Event objects to an SLayer object. You can also group SToken or SSpan coming from Event objects contained in several Tier objects to the same SLayer object.",
+        false));
+    this.addProperty(new PepperModuleProperty<String>(PROP_URI_ANNOTATION, String.class,
+        "With the property salt.URIAnnotation, you can mark a Tier object containing annotations, which are references in URI notation and are not simple String values. When mapping the data to Salt the resulting SAnnotation.sValue objects will be typed as URI values, so that each interpreting tool can interpret them as references.",
+        false));
+    this.addProperty(new PepperModuleProperty<String>(PROP_SALT_SEMANTICS_POS, String.class,
+        "You can influence the creation of objects in Salt to have a more semantic typing when mapping data to Salt. Here we provide three properties which can be used for a closer definition or typing of SAnnotation, SToken or SSpan objects conform to ISOCat1. This can be important in the case of a further processing with Pepper. Some modules exist, which only can deal with semantical enriched data for example they need a special kind of annotation like part-of-speech for their processing.",
+        false));
+    this.addProperty(new PepperModuleProperty<String>(PROP_SALT_SEMANTICS_LEMMA, String.class,
+        "You can influence the creation of objects in Salt to have a more semantic typing when mapping data to Salt. Here we provide three properties which can be used for a closer definition or typing of SAnnotation, SToken or SSpan objects conform to ISOCat1. This can be important in the case of a further processing with Pepper. Some modules exist, which only can deal with semantical enriched data for example they need a special kind of annotation like part-of-speech for their processing.",
+        false));
+    this.addProperty(new PepperModuleProperty<String>(PROP_SALT_SEMANTICS_WORD, String.class,
+        "You can influence the creation of objects in Salt to have a more semantic typing when mapping data to Salt. Here we provide three properties which can be used for a closer definition or typing of SAnnotation, SToken or SSpan objects conform to ISOCat1. This can be important in the case of a further processing with Pepper. Some modules exist, which only can deal with semantical enriched data for example they need a special kind of annotation like part-of-speech for their processing.",
+        false));
+    this.addProperty(
+        new PepperModuleProperty<Boolean>(PROP_CLEAN_MODEL, Boolean.class, "....", false, false));
+    this.addProperty(new PepperModuleProperty<Boolean>(PROP_TRIM_EVENTS, Boolean.class,
+        "When this property is true, all events in exmaralda are trimmed before they are mapped to Salt. That means for instance trailing blanks are removed. ",
+        false, false));
+    this.addProperty(new PepperModuleProperty<Boolean>(PROP_PARSE_NAMESPACE, Boolean.class,
+        "Parse the qualified annotation name including the namespace (e.g. myns::annoname) from the category name.",
+        false, false));
+    this.addProperty(new PepperModuleProperty<Boolean>(PROP_MAP_TIMELINE, Boolean.class,
+        "Import the EXMARaLDA timeline as Salt STimeline", true, false));
+    this.addProperty(PepperModuleProperty.create().withName(PROP_MAP_DESCRIPTIONS)
+        .withType(Boolean.class).withDescription("Map tiers of the type \\\"Description\\\"")
+        .withDefaultValue(true).build());
+    this.addProperty(
+        PepperModuleProperty.create().withName(PROP_MAP_SPEAKER_METADATA).withType(Boolean.class)
+            .withDescription("Import the metadata of the speaker").withDefaultValue(true).build());
+    this.addProperty(PepperModuleProperty.create().withName(PROP_TEXT_NAME_WITH_NAMESPACE)
+        .withType(Boolean.class)
+        .withDescription(
+            "If true, texts will get qualified names with speaker name as namespace. If false, no namespace is used.")
+        .withDefaultValue(false).build());
 
-	public String getTierMerge() {
-		return ((String) this.getProperty(PROP_TIERMERGE).getValue());
-	}
+    this.addProperty(
+        PepperModuleProperty.create().withName(PROP_STRICT_FILE_TYPE).withType(Boolean.class)
+            .withDescription("Restrict importing files to having the file ending '.exb'.")
+            .withDefaultValue(true).build());
+  }
 
-	public String getURIAnnotation() {
-		return ((String) this.getProperty(PROP_URI_ANNOTATION).getValue());
-	}
+  @Deprecated
+  public Set<String> getTokenTiers() {
+    String prop = ((String) this.getProperty(PROP_TOKEN_TIER).getValue());
+    Set<String> tokenTiers = new LinkedHashSet<String>();
+    if ((prop != null) && (!prop.isEmpty())) {
+      if (prop.startsWith("{")) {
+        prop = prop.replace("{", "").replace("}", "");
+        String[] splitted = prop.split(",");
+        for (String s : splitted) {
+          tokenTiers.add(s.trim());
+        }
+      } else {
+        tokenTiers.add(prop.trim());
+      }
+    }
+    return (tokenTiers);
+  }
 
-	public String getPOS() {
-		return ((String) this.getProperty(PROP_SALT_SEMANTICS_POS).getValue());
-	}
+  public String getTokenSeparator() {
+    return ((String) this.getProperty(PROP_TOKENSEP).getValue());
+  }
 
-	public String getLemma() {
-		return ((String) this.getProperty(PROP_SALT_SEMANTICS_LEMMA).getValue());
-	}
+  public String getTierMerge() {
+    return ((String) this.getProperty(PROP_TIERMERGE).getValue());
+  }
 
-	public String getWord() {
-		return ((String) this.getProperty(PROP_SALT_SEMANTICS_WORD).getValue());
-	}
+  public String getURIAnnotation() {
+    return ((String) this.getProperty(PROP_URI_ANNOTATION).getValue());
+  }
 
-	public Boolean getCleanModel() {
-		return ((Boolean) this.getProperty(PROP_CLEAN_MODEL).getValue());
-	}
+  public String getPOS() {
+    return ((String) this.getProperty(PROP_SALT_SEMANTICS_POS).getValue());
+  }
 
-	public Boolean isTrimEvents() {
-		return ((Boolean) this.getProperty(PROP_TRIM_EVENTS).getValue());
-	}
-	
-	public Boolean isParseNamespace() {
-		return ((Boolean) this.getProperty(PROP_PARSE_NAMESPACE).getValue());
-	}
-	
-	public Boolean isMapTimeline() {
-		return ((Boolean) this.getProperty(PROP_MAP_TIMELINE).getValue());
-	}
-	
-	public Boolean isMapDescriptions() {
-		return ((Boolean) this.getProperty(PROP_MAP_DESCRIPTIONS).getValue());
-	}
-	
-	public Boolean isMapSpeakerMetadata() {
-		return ((Boolean) this.getProperty(PROP_MAP_SPEAKER_METADATA).getValue());
-	}
-	
-	
-	/**
-	 * String for regex for for tier to layer mapping
-	 */
-	private static final String TIER_NAME_DESC = "(_|-|[A-Z]|[a-z]|[0-9])+";
-	/**
-	 * String for regex for for tier to layer mapping
-	 */
-	private static final String SIMPLE_TIER_LIST_DESC = "\\{" + TIER_NAME_DESC + "(,\\s?" + TIER_NAME_DESC + ")*" + "\\}";
-	/**
-	 * String for regex for for tier to layer mapping
-	 */
-	private static final String LAYER_NAME_DESC = "(_|-|[A-Z]|[a-z]|[0-9])+";
-	/**
-	 * String for regex for for tier to layer mapping
-	 */
-	private static final String SIMPLE_LAYER_DESC = "\\{" + LAYER_NAME_DESC + SIMPLE_TIER_LIST_DESC + "\\}";
+  public String getLemma() {
+    return ((String) this.getProperty(PROP_SALT_SEMANTICS_LEMMA).getValue());
+  }
 
-	/**
-	 * Checks the given properties, if the necessary ones are given and creates
-	 * the data-structures being needed to store the properties. Throws an
-	 * exception, if the needed properties are not there.
-	 * 
-	 * @return Relates the name of the tiers to the layers, to which they shall
-	 *         be append.
-	 */
-	public Map<String, SLayer> getTier2SLayers() {
-		Hashtable<String, SLayer> tierNames2SLayers = new Hashtable<String, SLayer>();
+  public String getWord() {
+    return ((String) this.getProperty(PROP_SALT_SEMANTICS_WORD).getValue());
+  }
 
-		// tiers to SLayer-objects
-		String tier2SLayerStr = (String) this.getProperty(PROP_LAYERS_SMALL).getValue();
-		if (tier2SLayerStr == null) {
-			tier2SLayerStr = (String) this.getProperty(PROP_LAYERS_BIG).getValue();
-		}
+  public Boolean getCleanModel() {
+    return ((Boolean) this.getProperty(PROP_CLEAN_MODEL).getValue());
+  }
 
-		if ((tier2SLayerStr != null) && (!tier2SLayerStr.trim().isEmpty())) {
-			// if a tier to layer mapping is given
-			// check if number of closing brackets is identical to number of
-			// opening brackets
-			char[] tier2SLayerChar = tier2SLayerStr.toCharArray();
-			int numberOfOpeningBrackets = 0;
-			int numberOfClosingBrackets = 0;
-			for (int i = 0; i < tier2SLayerChar.length; i++) {
-				if (tier2SLayerChar.equals('{'))
-					numberOfOpeningBrackets++;
-				else if (tier2SLayerChar.equals('}'))
-					numberOfClosingBrackets++;
-			}
-			if (numberOfClosingBrackets != numberOfOpeningBrackets) {
-				throw new PepperModuleException("Cannot import the given data, because property file contains a corrupt value for property '" + EXMARaLDAImporterProperties.PROP_LAYERS_BIG + "'. Please check the breckets you used.");
-			}
-			tier2SLayerStr = tier2SLayerStr.replace(" ", "");
-			Pattern pattern = Pattern.compile(SIMPLE_LAYER_DESC, Pattern.CASE_INSENSITIVE);
-			Matcher matcher = pattern.matcher(tier2SLayerStr);
-			while (matcher.find()) {
-				// find all simple layer descriptions
-				String[] tierNames = null;
-				String tierNameList = null;
-				Pattern pattern1 = Pattern.compile(SIMPLE_TIER_LIST_DESC, Pattern.CASE_INSENSITIVE);
-				Matcher matcher1 = pattern1.matcher(matcher.group());
-				while (matcher1.find()) {
-					// find all tier lists
-					tierNameList = matcher1.group();
-					tierNames = tierNameList.replace("}", "").replace("{", "").split(",");
-				}
-				String sLayerName = matcher.group().replace(tierNameList, "").replace("}", "").replace("{", "");
-				SLayer sLayer = SaltFactory.createSLayer();
-				sLayer.setName(sLayerName);
-				for (String tierName : tierNames) {
-					// put all tiernames in table to map them to the layer
-					tierNames2SLayers.put(tierName, sLayer);
-				}
-			}
+  public Boolean isTrimEvents() {
+    return ((Boolean) this.getProperty(PROP_TRIM_EVENTS).getValue());
+  }
 
-			if (tierNames2SLayers.size() == 0) {
-				logger.warn("[EXMARaLDAImporter] It seems as if there is a syntax failure in the given special-param file in property '" + EXMARaLDAImporterProperties.PROP_LAYERS_BIG + "'. A value is given, but the layers to named could not have been extracted.");
-			}
-		}
-		return (tierNames2SLayers);
-	}
-	
-	public boolean qualifyTextNames() {
-		return (Boolean) getProperty(PROP_TEXT_NAME_WITH_NAMESPACE).getValue();
-	}
+  public Boolean isParseNamespace() {
+    return ((Boolean) this.getProperty(PROP_PARSE_NAMESPACE).getValue());
+  }
+
+  public Boolean isMapTimeline() {
+    return ((Boolean) this.getProperty(PROP_MAP_TIMELINE).getValue());
+  }
+
+  public Boolean isMapDescriptions() {
+    return ((Boolean) this.getProperty(PROP_MAP_DESCRIPTIONS).getValue());
+  }
+
+  public Boolean isMapSpeakerMetadata() {
+    return ((Boolean) this.getProperty(PROP_MAP_SPEAKER_METADATA).getValue());
+  }
+
+
+  /**
+   * String for regex for for tier to layer mapping
+   */
+  private static final String TIER_NAME_DESC = "(_|-|[A-Z]|[a-z]|[0-9])+";
+  /**
+   * String for regex for for tier to layer mapping
+   */
+  private static final String SIMPLE_TIER_LIST_DESC =
+      "\\{" + TIER_NAME_DESC + "(,\\s?" + TIER_NAME_DESC + ")*" + "\\}";
+  /**
+   * String for regex for for tier to layer mapping
+   */
+  private static final String LAYER_NAME_DESC = "(_|-|[A-Z]|[a-z]|[0-9])+";
+  /**
+   * String for regex for for tier to layer mapping
+   */
+  private static final String SIMPLE_LAYER_DESC =
+      "\\{" + LAYER_NAME_DESC + SIMPLE_TIER_LIST_DESC + "\\}";
+
+  /**
+   * Checks the given properties, if the necessary ones are given and creates the data-structures
+   * being needed to store the properties. Throws an exception, if the needed properties are not
+   * there.
+   * 
+   * @return Relates the name of the tiers to the layers, to which they shall be append.
+   */
+  public Map<String, SLayer> getTier2SLayers() {
+    Hashtable<String, SLayer> tierNames2SLayers = new Hashtable<String, SLayer>();
+
+    // tiers to SLayer-objects
+    String tier2SLayerStr = (String) this.getProperty(PROP_LAYERS_SMALL).getValue();
+    if (tier2SLayerStr == null) {
+      tier2SLayerStr = (String) this.getProperty(PROP_LAYERS_BIG).getValue();
+    }
+
+    if ((tier2SLayerStr != null) && (!tier2SLayerStr.trim().isEmpty())) {
+      // if a tier to layer mapping is given
+      // check if number of closing brackets is identical to number of
+      // opening brackets
+      char[] tier2SLayerChar = tier2SLayerStr.toCharArray();
+      int numberOfOpeningBrackets = 0;
+      int numberOfClosingBrackets = 0;
+      for (int i = 0; i < tier2SLayerChar.length; i++) {
+        if (tier2SLayerChar.equals('{'))
+          numberOfOpeningBrackets++;
+        else if (tier2SLayerChar.equals('}'))
+          numberOfClosingBrackets++;
+      }
+      if (numberOfClosingBrackets != numberOfOpeningBrackets) {
+        throw new PepperModuleException(
+            "Cannot import the given data, because property file contains a corrupt value for property '"
+                + EXMARaLDAImporterProperties.PROP_LAYERS_BIG
+                + "'. Please check the breckets you used.");
+      }
+      tier2SLayerStr = tier2SLayerStr.replace(" ", "");
+      Pattern pattern = Pattern.compile(SIMPLE_LAYER_DESC, Pattern.CASE_INSENSITIVE);
+      Matcher matcher = pattern.matcher(tier2SLayerStr);
+      while (matcher.find()) {
+        // find all simple layer descriptions
+        String[] tierNames = null;
+        String tierNameList = null;
+        Pattern pattern1 = Pattern.compile(SIMPLE_TIER_LIST_DESC, Pattern.CASE_INSENSITIVE);
+        Matcher matcher1 = pattern1.matcher(matcher.group());
+        while (matcher1.find()) {
+          // find all tier lists
+          tierNameList = matcher1.group();
+          tierNames = tierNameList.replace("}", "").replace("{", "").split(",");
+        }
+        String sLayerName =
+            matcher.group().replace(tierNameList, "").replace("}", "").replace("{", "");
+        SLayer sLayer = SaltFactory.createSLayer();
+        sLayer.setName(sLayerName);
+        for (String tierName : tierNames) {
+          // put all tiernames in table to map them to the layer
+          tierNames2SLayers.put(tierName, sLayer);
+        }
+      }
+
+      if (tierNames2SLayers.size() == 0) {
+        logger.warn(
+            "[EXMARaLDAImporter] It seems as if there is a syntax failure in the given special-param file in property '"
+                + EXMARaLDAImporterProperties.PROP_LAYERS_BIG
+                + "'. A value is given, but the layers to named could not have been extracted.");
+      }
+    }
+    return (tierNames2SLayers);
+  }
+
+  public boolean qualifyTextNames() {
+    return (Boolean) getProperty(PROP_TEXT_NAME_WITH_NAMESPACE).getValue();
+  }
+
+  public boolean getStrictFileType() {
+    return (Boolean) getProperty(PROP_STRICT_FILE_TYPE).getValue();
+  }
 }

--- a/src/test/java/org/corpus_tools/peppermodules/exmaralda/EXMARaLDAImporterTest.java
+++ b/src/test/java/org/corpus_tools/peppermodules/exmaralda/EXMARaLDAImporterTest.java
@@ -1,26 +1,25 @@
 /**
  * Copyright 2009 Humboldt-Universität zu Berlin, INRIA.
  *
- * Licensed under the Apache License, Version 2.0 (the "License");
- * you may not use this file except in compliance with the License.
- * You may obtain a copy of the License at
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
  *
- *       http://www.apache.org/licenses/LICENSE-2.0
+ * http://www.apache.org/licenses/LICENSE-2.0
  *
- * Unless required by applicable law or agreed to in writing, software
- * distributed under the License is distributed on an "AS IS" BASIS,
- * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
- * See the License for the specific language governing permissions and
- * limitations under the License.
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
  *
  *
  */
 package org.corpus_tools.peppermodules.exmaralda;
 
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.Assert.assertEquals;
 
 import java.io.File;
-
+import java.io.IOException;
 import org.corpus_tools.pepper.common.CorpusDesc;
 import org.corpus_tools.pepper.common.FormatDesc;
 import org.corpus_tools.pepper.common.ModuleFitness;
@@ -28,43 +27,105 @@ import org.corpus_tools.pepper.common.ModuleFitness.FitnessFeature;
 import org.corpus_tools.pepper.core.ModuleFitnessChecker;
 import org.corpus_tools.pepper.testFramework.PepperImporterTest;
 import org.corpus_tools.pepper.testFramework.PepperTestUtil;
-import org.corpus_tools.peppermodules.exmaralda.EXMARaLDAImporter;
 import org.corpus_tools.salt.SaltFactory;
-import org.corpus_tools.salt.util.SaltUtil;
+import org.corpus_tools.salt.common.SCorpusGraph;
 import org.eclipse.emf.common.util.URI;
 import org.junit.Before;
 import org.junit.Test;
 
 public class EXMARaLDAImporterTest extends PepperImporterTest {
 
-	@Before
-	public void setUp() throws Exception {
-		super.setFixture(new EXMARaLDAImporter());
-		super.getFixture().setSaltProject(SaltFactory.createSaltProject());
+  @Before
+  public void setUp() throws Exception {
+    super.setFixture(new EXMARaLDAImporter());
+    super.getFixture().setSaltProject(SaltFactory.createSaltProject());
 
-		// set formats to support
-		FormatDesc formatDef = new FormatDesc();
-		formatDef.setFormatName("EXMARaLDA");
-		formatDef.setFormatVersion("1.0");
-		addSupportedFormat(formatDef);
-	}
+    // set formats to support
+    FormatDesc formatDef = new FormatDesc();
+    formatDef.setFormatName("EXMARaLDA");
+    formatDef.setFormatVersion("1.0");
+    addSupportedFormat(formatDef);
+  }
 
-	@Test
-	public void testSetGetCorpusDefinition() {
-		// TODO somethong to test???
-		CorpusDesc corpDef = new CorpusDesc();
-		FormatDesc formatDef = new FormatDesc();
-		formatDef.setFormatName("EXMARaLDA");
-		formatDef.setFormatVersion("1.0");
-		corpDef.setFormatDesc(formatDef);
-	}
+  @Test
+  public void testSetGetCorpusDefinition() {
+    // TODO somethong to test???
+    CorpusDesc corpDef = new CorpusDesc();
+    FormatDesc formatDef = new FormatDesc();
+    formatDef.setFormatName("EXMARaLDA");
+    formatDef.setFormatVersion("1.0");
+    corpDef.setFormatDesc(formatDef);
+  }
 
-	@Test
-	public void whenSelfTestingModule_thenResultShouldBeTrue() {
-		final ModuleFitness fitness = new ModuleFitnessChecker(PepperTestUtil.createDefaultPepper()).selfTest(fixture);
-		assertThat(fitness.getFitness(FitnessFeature.HAS_SELFTEST)).isTrue();
-		assertThat(fitness.getFitness(FitnessFeature.HAS_PASSED_SELFTEST)).isTrue();
-		assertThat(fitness.getFitness(FitnessFeature.IS_IMPORTABLE_SEFTEST_DATA)).isTrue();
-		assertThat(fitness.getFitness(FitnessFeature.IS_VALID_SELFTEST_DATA)).isTrue();
-	}
+  @Test
+  public void whenSelfTestingModule_thenResultShouldBeTrue() {
+    final ModuleFitness fitness =
+        new ModuleFitnessChecker(PepperTestUtil.createDefaultPepper()).selfTest(fixture);
+    assertThat(fitness.getFitness(FitnessFeature.HAS_SELFTEST)).isTrue();
+    assertThat(fitness.getFitness(FitnessFeature.HAS_PASSED_SELFTEST)).isTrue();
+    assertThat(fitness.getFitness(FitnessFeature.IS_IMPORTABLE_SEFTEST_DATA)).isTrue();
+    assertThat(fitness.getFitness(FitnessFeature.IS_VALID_SELFTEST_DATA)).isTrue();
+  }
+
+  @Test
+  public void testStrictFileEnding() throws IOException {
+
+    File corpusPathFile = new File("./src/test/resources/testCorpora/specificFileEnding/");
+
+    URI corpusPath = URI.createFileURI(corpusPathFile.getCanonicalPath());
+
+
+    // creating and setting corpus definition
+    CorpusDesc corpDef = new CorpusDesc();
+    FormatDesc formatDef = new FormatDesc();
+    formatDef.setFormatName("EXMARaLDA");
+    formatDef.setFormatVersion("1.0");
+    corpDef.setFormatDesc(formatDef);
+    corpDef.setCorpusPath(corpusPath);
+    getFixture().setCorpusDesc(corpDef);
+
+    getFixture().getProperties().setPropertyValue(EXMARaLDAImporterProperties.PROP_STRICT_FILE_TYPE,
+        true);
+
+
+    // importing corpus
+    start();
+
+
+    SCorpusGraph cg = getFixture().getCorpusGraph();
+    assertEquals(1, cg.getCorpora().size());
+    assertEquals("specificFileEnding", cg.getCorpora().get(0).getName());
+    assertEquals(1, cg.getDocuments().size());
+    assertEquals("doc1", cg.getDocuments().get(0).getName());
+  }
+
+  @Test
+  public void testNonStrictFileEnding() throws IOException {
+
+    File corpusPathFile = new File("./src/test/resources/testCorpora/specificFileEnding/");
+
+    URI corpusPath = URI.createFileURI(corpusPathFile.getCanonicalPath());
+
+
+    // creating and setting corpus definition
+    CorpusDesc corpDef = new CorpusDesc();
+    FormatDesc formatDef = new FormatDesc();
+    formatDef.setFormatName("EXMARaLDA");
+    formatDef.setFormatVersion("1.0");
+    corpDef.setFormatDesc(formatDef);
+    corpDef.setCorpusPath(corpusPath);
+    getFixture().setCorpusDesc(corpDef);
+
+    getFixture().getProperties().setPropertyValue(EXMARaLDAImporterProperties.PROP_STRICT_FILE_TYPE,
+        false);
+
+    // importing corpus
+    start();
+
+
+    SCorpusGraph cg = getFixture().getCorpusGraph();
+    assertEquals(1, cg.getCorpora().size());
+    assertEquals("specificFileEnding", cg.getCorpora().get(0).getName());
+    assertEquals(2, cg.getDocuments().size());
+  }
 }

--- a/src/test/resources/testCorpora/specificFileEnding/doc1.exb
+++ b/src/test/resources/testCorpora/specificFileEnding/doc1.exb
@@ -1,0 +1,71 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- (c) http://www.rrz.uni-hamburg.de/exmaralda -->
+<basic-transcription>
+<head>
+<meta-information>
+	<project-name>doc1</project-name>
+	<transcription-name>fu001_2004_05</transcription-name>
+	<referenced-file url=""/>
+	<ud-meta-information><ud-information attribute-name="transcriptor">JK</ud-information><ud-information attribute-name="transcription-date">09.02.04</ud-information><ud-information attribute-name="transcription-modality">hand</ud-information><ud-information attribute-name="production-modality">Zusammenfassung</ud-information><ud-information attribute-name="topic">Entscheidungen</ud-information><ud-information attribute-name="sub-corpus">1.1</ud-information></ud-meta-information>
+	<comment></comment>
+	<transcription-convention></transcription-convention>
+</meta-information>
+<speakertable>
+	<speaker id="s0">
+		<abbreviation></abbreviation>
+		<sex value="m"/>
+		<languages-used></languages-used>
+		<l1></l1>
+		<l2></l2>
+		<ud-speaker-information>
+		<ud-information attribute-name="name">Andreeva</ud-information><ud-information attribute-name="first-name">Marianella</ud-information><ud-information attribute-name="birth-year">1982</ud-information><ud-information attribute-name="sex">f</ud-information><ud-information attribute-name="major-subject">Linguistik</ud-information><ud-information attribute-name="ctest">k.A.</ud-information><ud-information attribute-name="l1">bg</ud-information><ud-information attribute-name="l12"></ud-information><ud-information attribute-name="l2">en</ud-information><ud-information attribute-name="l3">de</ud-information><ud-information attribute-name="l4"></ud-information><ud-information attribute-name="l5"></ud-information><ud-information attribute-name="l6"></ud-information><ud-information attribute-name="l7"></ud-information><ud-information attribute-name="l2-duration">k.A.</ud-information><ud-information attribute-name="l3-duration">k.A.</ud-information><ud-information attribute-name="l4-duration"></ud-information><ud-information attribute-name="l5-duration"></ud-information><ud-information attribute-name="l6-duration"></ud-information><ud-information attribute-name="l7-duration"></ud-information><ud-information attribute-name="identifier">fu001</ud-information><ud-information attribute-name="l13"></ud-information><ud-information attribute-name="collection-date">09.02.04</ud-information><ud-information attribute-name="de-duration">k.A.</ud-information><ud-information attribute-name="subset">fu_2004_05</ud-information></ud-speaker-information>
+		<comment></comment>
+	</speaker>
+</speakertable></head>
+<basic-body>
+<common-timeline>
+<tli id="T1"/>
+<tli id="T2"/>
+<tli id="T3"/>
+<tli id="T4"/>
+<tli id="T5"/>
+<tli id="T6"/>
+<tli id="T7"/>
+<tli id="T8"/>
+<tli id="T9"/>
+<tli id="T10"/>
+</common-timeline>
+<tier id="TIE0" category="word" type="t" display-name="[word]">
+<event start="T1" end="T2">1.</event>
+<event start="T2" end="T3">Zusammenfassung</event>
+<event start="T3" end="T4">Unterscheidungen</event>
+<event start="T4" end="T5">treffen</event>
+<event start="T5" end="T6">(</event>
+<event start="T6" end="T7">George</event>
+<event start="T7" end="T8">A.</event>
+<event start="T8" end="T9">Miller</event>
+<event start="T9" end="T10">)</event>
+</tier><tier id="TIE1" category="lemma" type="t" display-name="[lemma]"><ud-tier-information><ud-information attribute-name="X-Generated-With">Decision Tree Tagger</ud-information><ud-information attribute-name="X-Generated-On">2007-11-01T17:12:13.309130</ud-information></ud-tier-information><event start="T1" end="T2">1.</event>
+<event start="T2" end="T3">Zusammenfassung</event>
+<event start="T3" end="T4">Unterscheidung</event>
+<event start="T4" end="T5">treffen</event>
+<event start="T5" end="T6">(</event>
+<event start="T6" end="T7">George</event>
+<event start="T7" end="T8">A.</event>
+<event start="T8" end="T9"><![CDATA[<unknown>]]></event>
+<event start="T9" end="T10">)</event>
+</tier><tier id="TIE2" category="pos" type="t" display-name="[pos]"><ud-tier-information><ud-information attribute-name="X-Generated-With">Decision Tree Tagger</ud-information><ud-information attribute-name="X-Generated-On">2007-11-01T17:12:13.289998</ud-information></ud-tier-information><event start="T1" end="T2">ADJA</event>
+<event start="T2" end="T3"><ud-information attribute-name="TestInfo">TestValue</ud-information>NN</event>
+<event start="T3" end="T4">NN</event>
+<event start="T4" end="T5">VVINF</event>
+<event start="T5" end="T6">$(</event>
+<event start="T6" end="T7">NE</event>
+<event start="T7" end="T8">NE</event>
+<event start="T8" end="T9">NE</event>
+<event start="T9" end="T10">$(</event>
+</tier><tier id="TIE3" speaker="s0" category="v" type="t" display-name=" [v]"><event start="T1" end="T3">Hallo</event>
+<event start="T3" end="T6">nochmal</event>
+</tier><tier id="TIE4" speaker="s0" category="v2" type="t" display-name=" [v2]"><event start="T1" end="T3">v2value1</event>
+<event start="T3" end="T6">v2value2</event>
+</tier></basic-body>
+</basic-transcription>

--- a/src/test/resources/testCorpora/specificFileEnding/doc2.xml
+++ b/src/test/resources/testCorpora/specificFileEnding/doc2.xml
@@ -1,0 +1,71 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- (c) http://www.rrz.uni-hamburg.de/exmaralda -->
+<basic-transcription>
+<head>
+<meta-information>
+	<project-name>doc2</project-name>
+	<transcription-name>fu001_2004_05</transcription-name>
+	<referenced-file url=""/>
+	<ud-meta-information><ud-information attribute-name="transcriptor">JK</ud-information><ud-information attribute-name="transcription-date">09.02.04</ud-information><ud-information attribute-name="transcription-modality">hand</ud-information><ud-information attribute-name="production-modality">Zusammenfassung</ud-information><ud-information attribute-name="topic">Entscheidungen</ud-information><ud-information attribute-name="sub-corpus">1.1</ud-information></ud-meta-information>
+	<comment></comment>
+	<transcription-convention></transcription-convention>
+</meta-information>
+<speakertable>
+	<speaker id="s0">
+		<abbreviation></abbreviation>
+		<sex value="m"/>
+		<languages-used></languages-used>
+		<l1></l1>
+		<l2></l2>
+		<ud-speaker-information>
+		<ud-information attribute-name="name">Andreeva</ud-information><ud-information attribute-name="first-name">Marianella</ud-information><ud-information attribute-name="birth-year">1982</ud-information><ud-information attribute-name="sex">f</ud-information><ud-information attribute-name="major-subject">Linguistik</ud-information><ud-information attribute-name="ctest">k.A.</ud-information><ud-information attribute-name="l1">bg</ud-information><ud-information attribute-name="l12"></ud-information><ud-information attribute-name="l2">en</ud-information><ud-information attribute-name="l3">de</ud-information><ud-information attribute-name="l4"></ud-information><ud-information attribute-name="l5"></ud-information><ud-information attribute-name="l6"></ud-information><ud-information attribute-name="l7"></ud-information><ud-information attribute-name="l2-duration">k.A.</ud-information><ud-information attribute-name="l3-duration">k.A.</ud-information><ud-information attribute-name="l4-duration"></ud-information><ud-information attribute-name="l5-duration"></ud-information><ud-information attribute-name="l6-duration"></ud-information><ud-information attribute-name="l7-duration"></ud-information><ud-information attribute-name="identifier">fu001</ud-information><ud-information attribute-name="l13"></ud-information><ud-information attribute-name="collection-date">09.02.04</ud-information><ud-information attribute-name="de-duration">k.A.</ud-information><ud-information attribute-name="subset">fu_2004_05</ud-information></ud-speaker-information>
+		<comment></comment>
+	</speaker>
+</speakertable></head>
+<basic-body>
+<common-timeline>
+<tli id="T1"/>
+<tli id="T2"/>
+<tli id="T3"/>
+<tli id="T4"/>
+<tli id="T5"/>
+<tli id="T6"/>
+<tli id="T7"/>
+<tli id="T8"/>
+<tli id="T9"/>
+<tli id="T10"/>
+</common-timeline>
+<tier id="TIE0" category="word" type="t" display-name="[word]">
+<event start="T1" end="T2">1.</event>
+<event start="T2" end="T3">Zusammenfassung</event>
+<event start="T3" end="T4">Unterscheidungen</event>
+<event start="T4" end="T5">treffen</event>
+<event start="T5" end="T6">(</event>
+<event start="T6" end="T7">George</event>
+<event start="T7" end="T8">A.</event>
+<event start="T8" end="T9">Miller</event>
+<event start="T9" end="T10">)</event>
+</tier><tier id="TIE1" category="lemma" type="t" display-name="[lemma]"><ud-tier-information><ud-information attribute-name="X-Generated-With">Decision Tree Tagger</ud-information><ud-information attribute-name="X-Generated-On">2007-11-01T17:12:13.309130</ud-information></ud-tier-information><event start="T1" end="T2">1.</event>
+<event start="T2" end="T3">Zusammenfassung</event>
+<event start="T3" end="T4">Unterscheidung</event>
+<event start="T4" end="T5">treffen</event>
+<event start="T5" end="T6">(</event>
+<event start="T6" end="T7">George</event>
+<event start="T7" end="T8">A.</event>
+<event start="T8" end="T9"><![CDATA[<unknown>]]></event>
+<event start="T9" end="T10">)</event>
+</tier><tier id="TIE2" category="pos" type="t" display-name="[pos]"><ud-tier-information><ud-information attribute-name="X-Generated-With">Decision Tree Tagger</ud-information><ud-information attribute-name="X-Generated-On">2007-11-01T17:12:13.289998</ud-information></ud-tier-information><event start="T1" end="T2">ADJA</event>
+<event start="T2" end="T3"><ud-information attribute-name="TestInfo">TestValue</ud-information>NN</event>
+<event start="T3" end="T4">NN</event>
+<event start="T4" end="T5">VVINF</event>
+<event start="T5" end="T6">$(</event>
+<event start="T6" end="T7">NE</event>
+<event start="T7" end="T8">NE</event>
+<event start="T8" end="T9">NE</event>
+<event start="T9" end="T10">$(</event>
+</tier><tier id="TIE3" speaker="s0" category="v" type="t" display-name=" [v]"><event start="T1" end="T3">Hallo</event>
+<event start="T3" end="T6">nochmal</event>
+</tier><tier id="TIE4" speaker="s0" category="v2" type="t" display-name=" [v2]"><event start="T1" end="T3">v2value1</event>
+<event start="T3" end="T6">v2value2</event>
+</tier></basic-body>
+</basic-transcription>

